include/fast_float/float_common.h

```diff
@@ -173,7 +173,7 @@ using parse_options = parse_options_t<char>;
 // rust style `try!()` macro, or `?` operator
 #define FASTFLOAT_TRY(x) { if (!(x)) return false; }
 
-#define FASTFLOAT_ENABLE_IF(...) typename std::enable_if<(__VA_ARGS__), int>::type = 0
```

Comment:
I do not mind this change per se, but can you explain why this was changed? Is there a reason?

Reply:
This is so that I can perform enable_if checks in this form: 
`typename = FASTFLOAT_ENABLE_IF(...)`
instead of 
`FASTFLOAT_ENABLE_IF(...) = 0`

The first form (default argument for a type parameter) is used in a template declaration in fast_float.h. Default arguments can only be specified once, so the template definition in parse_number.h cannot have it. With the first form, I can write the definition by just adding an extra typename, but with the second form I would have to repeat the FASTFLOAT_ENABLE_IF(...) again.